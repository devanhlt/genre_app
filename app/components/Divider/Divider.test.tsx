import React from "react"

import { ViewStyle } from "react-native"
import renderer from "react-test-renderer"
import { Divider } from "./Divider"

describe("<Divider />", () => {
  it("renders correctly horizontal type", () => {
    const tree = renderer.create(<Divider type="horizontal" />)
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly vertical type", () => {
    const tree = renderer.create(<Divider type="vertical" />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly vertical type and size 50", () => {
    const tree = renderer.create(<Divider type="vertical" size={50} />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly horizontal type and size 50", () => {
    const tree = renderer.create(<Divider type="horizontal" size={50} />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly style override", () => {
    const $styleOverride: ViewStyle = { backgroundColor: "red" }
    const tree = renderer.create(<Divider style={$styleOverride} />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with horizontal line", () => {
    const $styleOverride: ViewStyle = { backgroundColor: "red" }
    const tree = renderer.create(<Divider type="horizontal" style={$styleOverride} line />)
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly with vertical line", () => {
    const $styleOverride: ViewStyle = { backgroundColor: "red" }
    const tree = renderer.create(<Divider type="vertical" style={$styleOverride} line />)
    expect(tree).toMatchSnapshot()
  })
})
