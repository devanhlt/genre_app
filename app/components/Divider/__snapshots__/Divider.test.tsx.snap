// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> renders correctly horizontal type 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      {
        "height": 10,
      },
      false,
      undefined,
    ]
  }
/>
`;

exports[`<Divider /> renders correctly horizontal type and size 50 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      {
        "height": 50,
      },
      false,
      undefined,
    ]
  }
/>
`;

exports[`<Divider /> renders correctly style override 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      {
        "height": 10,
      },
      false,
      {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;

exports[`<Divider /> renders correctly vertical type 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      false,
      {
        "width": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Divider /> renders correctly vertical type and size 50 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      false,
      {
        "width": 50,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Divider /> renders correctly with horizontal line 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      {
        "height": 10,
      },
      false,
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#A1ABBB",
          "left": "50%",
          "position": "absolute",
          "top": "50%",
        },
        {
          "height": 1,
          "marginStart": -75,
          "marginTop": -1,
          "width": 150,
        },
        false,
      ]
    }
  />
</View>
`;

exports[`<Divider /> renders correctly with vertical line 1`] = `
<View
  style={
    [
      {
        "flexGrow": 0,
        "flexShrink": 0,
      },
      false,
      {
        "width": 10,
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#A1ABBB",
          "left": "50%",
          "position": "absolute",
          "top": "50%",
        },
        false,
        {
          "height": 50,
          "marginStart": -1,
          "marginTop": -25,
          "width": 1,
        },
      ]
    }
  />
</View>
`;
